lib/maze/server.rb

```diff
@@ -35,26 +28,32 @@ class << self
       # @return [String] The UUID attached to all command requests for this session
       attr_reader :command_uuid
 
-      # The intended HTTP status code on a successful request
+      # Sets the status code generator for the HTTP verb given.  If no verb is given then the
+      # generator will be shared across all allowable HTTP verbs.
       #
-      # @return [Integer] The HTTP status code, defaults to 200
-      def status_code(verb=nil)
-        if @status_override_verb
-          override_status = @status_override_verb.eql?(verb)
-        else
-          override_status = true
+      # @param verb [String] HTTP verb
+      def set_status_code_generator(generator, verb = nil)
+        @generators ||= {}
+        Array(verb || ALLOWED_HTTP_VERBS).each do |verb|
+          old = @generators[verb]
+          @generators[verb] = generator
+
+          # Close the old generator unless it's still being used by another verb
+          old&.close unless @generators.value?(old)
         end
+      end
 
-        if override_status
-          code = @status_code ||= 200
-          if reset_status_code
-            @status_code = 200
-            @status_override_verb = nil
-          end
+      # The intended HTTP status code on a successful request
+      #
+      # @param verb [String] HTTP verb for which the status code is wanted
+      #
+      # @return [Integer] The HTTP status code for the verb given
+      def status_code(verb)
+        if @generators[verb].nil? || @generators[verb].empty?
```

Comment:
As above; I don't think `empty?` could ever return true

```suggestion
        if @generators[verb].nil?
```

Reply:
Assuming you rename `empty?` &rarr; `closed?`:

```suggestion
        if @generators[verb].nil? || @generators[verb].closed?
```